Source/WebCore/dom/ObservableEventTarget.cpp

```diff
@@ -0,0 +1,113 @@
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+// https://wicg.github.io/observable/#event-target-integration
+
+#include "config.h"
+#include "ObservableEventTarget.h"
+
+#include "AddEventListenerOptions.h"
+#include "EventListener.h"
+#include "EventTarget.h"
+#include "JSEvent.h"
+#include "ObservableEventListenerOptions.h"
+#include "ScriptExecutionContext.h"
+#include "SubscribeOptions.h"
+#include "Subscriber.h"
+#include "SubscriberCallback.h"
+#include <wtf/WeakPtr.h>
+#include <wtf/text/AtomString.h>
+
+namespace WebCore {
+
+class ObservableEventListener final : public EventListener {
+public:
+    static Ref<ObservableEventListener> create(Ref<Subscriber>&& subscriber)
+    {
+        return adoptRef(*new ObservableEventListener(WTFMove(subscriber)));
+    }
+
+    void handleEvent(ScriptExecutionContext& context, Event& event) override
+    {
+        auto* globalObject = context.globalObject();
+        if (!globalObject)
+            return;
+
+        m_subscriber->next(toJS(globalObject, JSC::jsCast<JSDOMGlobalObject*>(globalObject), event));
+    }
+
+private:
+    ObservableEventListener(Ref<Subscriber>&& subscriber)
+        : EventListener(EventListener::CPPEventListenerType)
+        , m_subscriber(WTFMove(subscriber))
+    {
+    }
+
+    Ref<Subscriber> m_subscriber;
+};
+
+class SubscriberCallbackEventTarget final : public SubscriberCallback {
+public:
+    static Ref<SubscriberCallbackEventTarget> create(ScriptExecutionContext& context, EventTarget& eventTarget, const AtomString& eventType, const ObservableEventListenerOptions& options)
+    {
+        return adoptRef(*new SubscriberCallbackEventTarget(context, eventTarget, eventType, options));
+    }
+
+    CallbackResult<void> handleEvent(Subscriber& subscriber) final
+    {
+        if (subscriber.protectedSignal()->aborted())
+            return { };
+
+        if (RefPtr eventTarget = m_eventTarget.get()) {
+            AddEventListenerOptions addEventListenerOptions = { /* capture */ m_options.capture, /* passive */ m_options.passive, /* once */ false, /* signal */ subscriber.protectedSignal() };
```

Comment:
Yep, wasn't a fan either. But `AddEventListenerOptions` is not a struct where can just `.capture =`, because the struct inherits another one.

I saw this pattern of doing `/* XYZ */` [elsewhere as well](https://github.com/WebKit/WebKit/blob/da8e66a4fd1c8a97a7e7ec9b02c07cfb3dbb7c6d/Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp#L800).